// loopback4-authentication/dist/strategies/passport/passport-resource-owner-password/oauth2-resource-owner-password-grant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Oauth2ResourceOwnerPassword = void 0;
const tslib_1 = require("tslib");
const passport = tslib_1.__importStar(require("passport"));
var Oauth2ResourceOwnerPassword;
(function (Oauth2ResourceOwnerPassword) {
    class Strategy extends passport.Strategy {
        constructor(options, verify) {
            super();
            if (verify) {
                this.passReqToCallback = options.passReqToCallback;
                this.verify = verify;
            }
            else {
                this.passReqToCallback = false;
                this.verify = options;
            }
            this.name = 'oauth2-resource-owner-password';
        }
        authenticate(req, options) {
            var _a, _b, _c;
            if (
            /* eslint-disable @typescript-eslint/prefer-optional-chain */
            !req.body ||
                !((_a = req.body) === null || _a === void 0 ? void 0 : _a['client_id']) ||
                !((_b = req.body) === null || _b === void 0 ? void 0 : _b['username']) ||
                !((_c = req.body) === null || _c === void 0 ? void 0 : _c['password'])) {
                this.fail();
                return;
            }
            const clientId = req.body['client_id'];
            const clientSecret = req.body['client_secret'];
            const username = req.body['username'];
            const password = req.body['password'];
            const verified = (err, client, user) => {
                if (err) {
                    this.error(err);
                    return;
                }
                if (!client) {
                    this.fail();
                    return;
                }
                if (!user) {
                    this.fail();
                    return;
                }
                this.success(user);
            };
            if (this.passReqToCallback) {
                this.verify(req, clientId, clientSecret, username, password, verified);
            }
            else {
                this.verify(clientId, clientSecret, username, password, verified);
            }
        }
    }
    Oauth2ResourceOwnerPassword.Strategy = Strategy;
})(Oauth2ResourceOwnerPassword = exports.Oauth2ResourceOwnerPassword || (exports.Oauth2ResourceOwnerPassword = {}));
